doc-surrealdb_versioned_docs/version-latest/surrealql/statements/define/access/index.mdx

```diff
@@ -67,6 +67,18 @@ DEFINE ACCESS account ON DATABASE TYPE RECORD
 ;
 ```
 
+## With `AUTHENTICATE` clause
+
+The authenticate clause can be used to change the record identifier returned by the `SIGNIN` and `SIGNUP` clauses or replace the identifier provided in the token when authenticating `WITH JWT` in the context of `DEFINE ACCESS ... TYPE RECORD`. 
+
+In the context of [`DEFINE ACCESS ... TYPE RECORD`](/docs/surrealdb/surrealql/statements/define/access/record), the `AUTHENTICATE` clause is always executed across signin, signup and token authentication. This clause can be used to log or stop authentication attempts from record users, as it is always executed across signin, signup and token authentication. The clause always expects a record identifier to be returned and will otherwise fail.
+
+When used in a [`DEFINE ACCESS ... TYPE JWT`](/docs/surrealdb/surrealql/statements/define/access/jwt), the `AUTHENTICATE` clause is used to validate the token claims and can be used to log or stop authentication attempts. The clause always expects a boolean value to be returned and will otherwise fail.
```

Comment:
Not returning anything (aka `NONE`) will not result in a failure either

Reply:
Yeah. The only way this clause fails is if the query fails or a an error is thrown.

Reply:
Or when false is returned right?

Reply:
Oh my bad, when anything is returned the query also fails. This includes `false`, yes.

Reply:
I would write: "The clause expects nothing to be returned and will otherwise fail with a generic error. The `THROW` statement can be called to return a custom error to the end user.".